client/src/components/AuditBoard/AuditBoardView.test.tsx

```diff
@@ -535,5 +536,71 @@ describe('AuditBoardView', () => {
         await screen.findByText('Audit Ballot Selections')
       })
     })
+
+    it('clears all choices when "Ballot Not Found" is selected, even if previous choices were almost submitted', async () => {
```

Comment:
There are ways to write the choice clearing code such that you pass the above test (clears all choices when "Ballot Not Found" is selected) but not this one (clears all choices when "Ballot Not Found" is selected, **even if previous choices were almost submitted**) and vice versa

Reply:
Nice!